src/box/box.cc

```diff
@@ -4434,6 +4434,33 @@ box_process_fetch_snapshot(struct iostream *io,
 			  "wal_mode = 'none'");
 	}
 
+	/*
+	 * Find checkpoint for checkpoint join. If replica didn't request
+	 * specific one, take the newest one. Initialize checkpoint cursor
+	 * with chosen checkpoint and use its vclock for WAL GC consumer.
+	 * If requested checkpoint is not found, raise an error.
+	 */
+	struct checkpoint_cursor cursor;
+	struct checkpoint_cursor *cursor_ptr = NULL;
+	struct gc_checkpoint *checkpoint = NULL;
+	const struct vclock *gc_vclock = instance_vclock;
+	if (req.is_checkpoint_join) {
```

Comment:
This new branch and the one below should be covered in CE tests.

Reply:
Thanks, done!